src/fpm_source_parsing.f90

```diff
@@ -655,5 +612,100 @@ function parse_sequence(string,t1,t2,t3,t4) result(found)
 
 end function parse_sequence
 
+! USE [, intrinsic] :: module_name [, only: only_list]
+! USE [, non_intrinsic] :: module_name [, only: only_list]
+subroutine parse_use_statement(f_filename,i,line,use_stmt,is_intrinsic,module_name,error)
+    character(*), intent(in) :: f_filename,line
+    integer, intent(in) :: i ! line number
+    logical, intent(out) :: use_stmt,is_intrinsic
+    character(:), allocatable, intent(out) :: module_name
+    type(error_t), allocatable, intent(out) :: error
+
+    character(15), parameter :: INTRINSIC_NAMES(*) =  &
+                                 ['iso_c_binding  ', &
+                                  'iso_fortran_env', &
+                                  'ieee_arithmetic', &
+                                  'ieee_exceptions', &
+                                  'ieee_features  ', &
+                                  'omp_lib        ']
+
+    character(len=:), allocatable :: lowercase,temp_string
+    integer :: colons,intr,nonintr,j,stat
+    logical :: has_intrinsic_name
+
+    use_stmt      = .false.
+    is_intrinsic  = .false.
+    if (len_trim(line)<=0) return
+
+    ! Preprocess: lowercase, remove heading spaces
+    lowercase = lower(trim(adjustl(line)))
```

Comment:
Do we absolutely need to repeat this preprocessing (adjustl, trim, lower)? We already do this in the calling function. I see why you might want this function to be robust to inputs, but for the sake of efficiency I would suggest that we don't repeat this work if possible since it is already quite a significant part of the parsing cost. Great work with the rest of the PR! 👍

Reply:
Exactly @LKedward, I had gone for safety. Gotten rid of that, now just doublechecking that the first character is not a space (should be fast). Thank you for reviewing!